src/client/logging/transports.ts

```diff
@@ -71,6 +73,33 @@ export function getConsoleTransport(formatter: logform.Format): Transport {
     });
 }
 
+export function setPythonOutputChannelTransport(outputChannel: IOutputChannel) {
+    pythonOutputChannel = outputChannel;
+}
+
+class PythonOutputChannelTransport extends Transport {
+    // tslint:disable-next-line: no-any
+    constructor(options?: any) {
+        super(options);
+    }
+    // tslint:disable-next-line: no-any
+    public log?(info: { level: string; message: string; [formattedMessage]: string }, next: () => void): any {
+        setImmediate(() => this.emit('logged', info));
+        pythonOutputChannel.appendLine(info[formattedMessage] || info.message);
+        if (next) {
+            next();
+        }
+    }
+}
+
+// Create a Python output channel targeting transport ;that can be added to a winston logger.
+export function getPythonOutputChannelTransport(formatter: logform.Format) {
+    return new PythonOutputChannelTransport({
+        // We minimize customization.
+        format: formatter
+    });
+}
```

Comment:
As noted above, I recommend avoiding a global output channel.  Instead the transport should use a provided channel when created:

```suggestion
class PythonOutputChannelTransport extends Transport {
    // tslint:disable-next-line: no-any
    constructor(private readonly channel: IOutputChannel, options?: any) {
        super(options);
    }
    // tslint:disable-next-line: no-any
    public log?(info: { level: string; message: string; [formattedMessage]: string }, next: () => void): any {
        setImmediate(() => this.emit('logged', info));
        this.channel.appendLine(info[formattedMessage] || info.message);
        if (next) {
            next();
        }
    }
}

// Create a Python output channel targeting transport ;that can be added to a winston logger.
export function getPythonOutputChannelTransport(config: ChannelConfig, formatter: logform.Format) {
    return new PythonOutputChannelTransport(
        config.channel,
        {
            // We minimize customization.
            format: formatter
        }
    );
}
```